source-code-git/fs/src/utilities/toJson.ts

```diff
@@ -37,6 +37,7 @@ export async function toJson(args: {
 	 * TODO: to make it more clear what it does? `resolveFrom` might be ambigous.
 	 */
 	resolveFrom: string
+	encodeOutput?: "utf-8" | "base64"
```

Comment:
Can we always use `base64` or `Uint8Array` for `toJson` and `fromJson`? 

Enforcing an encoding will eventually break (?). What about a filesystem containing binary (base64) and non-binary data (utf-8)? Enforcing one encoding type will break a roundtrip because all files will become either binary or a string. 

I am wondering if we can set everything to be base64 or `Uint8Array` and don't define an encoding at all.

Reply:
Hmm you're right, I didn't realize reading binary data into a `string` would corrupt it. Perhaps too much C programming from me :P

`Uint8Array` would be a great solution for in-memory JSON representations, but would be significantly less efficient than `base64` when writing the JSON to a text file (at least the way I'm imagining it).  I think `base64` provides a nice middle ground so I'll enforce that instead.

Reply:
> Perhaps too much C programming from me :P 

Haha. One would assume that C would be harsher in this case than JavaScript :D 

If `base64` works for roundtrips of binary and non-binary data, let's choose base64 and get rid of the encoding argument.

Reply:
Edit: Just saw your reply on Discord. I replied there https://discord.com/channels/897438559458430986/1101312651637821520/1101974539367354541

Reply:
> If base64 works for roundtrips of binary and non-binary data

In our case, since `memoryFs` is implemented using `Map<string, string>`, I'm not sure if round tripping binary data would work. I think we may need to change the implementation to `Map<string, Uint8Array` or something else.

Reply:
I belive that when we call node's `readFile` with the `encoding` option set to `utf-8`, any invalid unicode characters are replaced with <?>, corrupting any binary data. If we wanted to be able to roundtrip binary and string data, we would need to base64 encode the raw `Buffer` objects, decode them, and then write them back to the filesystem as `Buffer` instead of `string`.

Reply:
ah, looks like what was missing was to specify `{ encoding: 'binary' }` for `readFile`. This should be doable without using `Uint8Array` then.